.circleci/Makefile

```diff
@@ -72,6 +74,8 @@ export GENERATED_FILE_HEADER
 # it them moves that file to $@. This makes is an atomic operation, so if it fails
 # make doesn't consider a half-baked file up to date.
 define GEN_CONFIG
+	@yq -i ".references.environment.GO_IMAGE = \"docker.mirror.hashicorp.services/cimg/go:$(GO_VERSION)\"" $(SOURCE_DIR)/executors/\@executors.yml
```

Comment:
Should we do a preflight check to make sure `yq` is installed, or:
- is this guaranteed to be around?
- "command not found" is sufficient

Reply:
It's guaranteed to be present in both the machine executor and docker images from CircleCI.